src/main/reactjs/src/components/publicTranslator/PublicTranslatorsGrid.tsx

```diff
@@ -24,32 +24,59 @@ export const PublicTranslatorsGrid = () => {
   const [showTable, setShowTable] = useState(false);
   const hasResults = translators.length > 0 && showTable;
   const hasNoResults = !hasResults && showTable;
+  const isLoading = status === APIResponseStatus.InProgress;
+
+  const renderSkeletons = () => (
+    <>
+      <Skeleton variant={SkeletonVariant.Text}>
+        <H1 className="public-homepage__filters__heading-title">
+          {t('filters.title')}
+        </H1>
+      </Skeleton>
+      <Skeleton className="full-max-width" variant={SkeletonVariant.Text}>
+        <Text className="public-homepage__filters__heading-description">
+          {t('note')}
+        </Text>
+      </Skeleton>
+      <Skeleton
+        className="full-max-width"
+        variant={SkeletonVariant.Rectangular}
+      >
+        <PublicTranslatorFilters setShowTable={setShowTable} />
+      </Skeleton>
+    </>
+  );
 
   return (
     <>
-      <Grid item>
-        <H1 data-testid="homepage__title-heading">{t('title')}</H1>
+      <Grid item className="public-homepage__grid-container__item-header">
+        <H1 data-testid="public-homepage__title-heading">{t('title')}</H1>
+        <HeaderSeparator />
         <Text>{t('description')}</Text>
       </Grid>
-      <Grid item>
-        <Paper elevation={3} className="homepage__filters">
-          <H1>{t('filters.title')}</H1>
-          <PublicTranslatorFilters setShowTable={setShowTable} />
+      <Grid item className="public-homepage__grid-container__item-filters">
+        <Paper elevation={3} className="public-homepage__filters">
+          {isLoading ? (
+            renderSkeletons()
+          ) : (
+            <>
+              <H1 className="public-homepage__filters__heading-title">
+                {t('filters.title')}
+              </H1>
+              <Text className="public-homepage__filters__heading-description">
+                {t('note')}
+              </Text>
+              <PublicTranslatorFilters setShowTable={setShowTable} />
+            </>
+          )}
```

Comment:
React Suspensen käyttäminen datan latauksen yhteydessä voisi olla kanssa yksi vaihtoehto. Sekin tarjoaa helpon tavan määrittää fallback-komponentti rendattavaksi kun data ei ole vielä ladattu.

Reply:
Kyllä, ajattelin itsekin samaa, mutta se taitaa olla tällä hetkellä experimental tilassa, toki voidaan käyttää jos halutaan, mutta se tulee virallisesti React 18 yhteydessä.

Reply:
Tosiaan ei tarvetta tässä yhteydessä, mutta hyvä pitää mielessä jos tuntuu että kaipaa tiukemmin strukturoitua tapaa reagoida eri lataus- ja virhetiloihin. Näin pienessä projektissa voidaan toki hyvin pärjätä ilmankin.